explorer/client/src/components/displaybox/DisplayBox.tsx

```diff
@@ -52,6 +86,11 @@ function DisplayBox({ display }: { display: string }) {
                     onError={handleImageFail}
                 />
             )}
+            {showAutoModNotice && (
+                <div className={styles.automod} id="modnotice">
+                    NFT image hidden
+                </div>
+            )}
```

Comment:
The different combinations of multiple boolean flags make this hard to reason about. You can remove all the `!showAutoModNotice` by early returning `NFT image hidden` div  if `showAutoModNotice === true`

Reply:
changed this, it requires some more code duplication but is clearer